app/code/Magento/Catalog/Model/ResourceModel/Category.php

```diff
@@ -715,7 +715,7 @@ public function getCategories($parent, $recursionLevel = 0, $sorted = false, $as
         /* @var $tree \Magento\Catalog\Model\ResourceModel\Category\Tree */
         $nodes = $tree->loadNode($parent)->loadChildren($recursionLevel)->getChildren();
 
-        $tree->addCollectionData(null, $sorted, $parent, $toLoad, true);
+        $tree->addCollectionData(null, $sorted, $parent, $toLoad, true, false);
```

Comment:
Instead of change, this `getCategories` method wouldn't make more sense to look at the CatalogUrlRewrite plugins and use CategoryRepository there to fetch the required categories?